### PR TITLE
nix{Flakes,Unstable}/update-1

### PR DESCRIPTION
**Pull request by eadwu**
###### Motivation for this change
No real point to having `sandbox-fallback` if it doesn't exist.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @

### CODE REVIEW

**Comment by dywedir** [comment]
Could you please also update `nixFlakes`?

File: pkgs/tools/package-management/nix/default.nix [hunk 1]
```diff
@@ -195,12 +195,12 @@ in rec {
...
     name = "nix-2.3${suffix}";
-    suffix = "pre20190712_aa82f8b";
+    suffix = "pre20190829_ebc4dae";
     src = fetchFromGitHub {
       owner = "NixOS";
       repo = "nix";
-      rev = "aa82f8b2d2a2c42f0d713e8404b668cef1a4b108";
-      hash = "sha256-MRY2CCjnTPSWIv0/aguZcg5U+DA+ODLKl9vjB/qXFpU=";
+      rev = "ebc4dae51761cb54c1db28cf4d328d3f2f55b758";
+      hash = "08rx6k15rrcjjdzmwzifvvsr6vpz0n5cccy9hv441fz5pvl7kyn9";
```
Comment (eadwu) [inline pkgs/tools/package-management/nix/default.nix]: How is the hash calculated here?

File: pkgs/tools/package-management/nix/default.nix [hunk 2]
```diff
@@ -181,12 +181,12 @@ in rec {
 
   nixUnstable = lib.lowPrio (callPackage common rec {
     name = "nix-2.3${suffix}";
-    suffix = "pre6779_324a5dc9";
+    suffix = "pre6779_84de821";
```
Comment (edolstra) [inline pkgs/tools/package-management/nix/default.nix]: Shouldn't 6779 be changed?

File: pkgs/tools/package-management/nix/default.nix (hunk 1 above)
Comment (edolstra) [reply pkgs/tools/package-management/nix/default.nix]: `hash` should be an SRI hash, which can be computed using `nix hash-file` or `nix to-sri`, e.g.
```
$ nix to-sri --type sha256 08rx6k15rrcjjdzmwzifvvsr6vpz0n5cccy9hv441fz5pvl7kyn9
sha256-yfp56L7lu0DIhskzxooF/26T9d4ufl5/k5LlXMI0PSM=
```

File: pkgs/tools/package-management/nix/default.nix (hunk 2 above)
Comment (eadwu) [reply pkgs/tools/package-management/nix/default.nix]: Where does that part of the suffix come from, looking at the source there doesn't seem to be a reference to `6779` anywhere in either commit?

Comment (edolstra) [reply pkgs/tools/package-management/nix/default.nix]: `git rev-list $revision | wc -l`

Comment (eadwu) [reply pkgs/tools/package-management/nix/default.nix]: Done, thanks for help.

**Pull request merged**